security-csrf/src/main/java/io/micronaut/security/csrf/resolver/FieldCsrfTokenResolver.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.security.csrf.resolver;
+
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.core.util.StringUtils;
+import io.micronaut.http.HttpRequest;
+import io.micronaut.http.ServerHttpRequest;
+import io.micronaut.http.server.filter.FilterBodyParser;
+import io.micronaut.security.csrf.CsrfConfiguration;
+import jakarta.inject.Singleton;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Resolves a CSRF token from a form-urlencoded body using the {@link ServerHttpRequest#byteBody()} API.
+ *
+ * @since 2.0.0
+ */
+@Requires(classes = HttpRequest.class)
+@Requires(property = "micronaut.security.csrf.token-resolvers.field.enabled", value = StringUtils.TRUE, defaultValue = StringUtils.TRUE)
```

Comment:
extract this property into a constant

Reply:
resolved via https://github.com/micronaut-projects/micronaut-security/pull/1826/commits/8b81613714ce6758decac40bf52d79d7d21ae033